src/test/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheckTest.java

```diff
@@ -371,4 +371,18 @@ public void testClearStatePackageName() throws Exception {
                 .isTrue();
     }
 
+    /**
+     * Test [ClassName]::new expression. Such expression will be processed as a
+     * normal declaration of a new instance. We need to make sure this does not
+     *
+     * @throws Exception when code tested throws exception
+     */
+    @Test
+    public void testLambdaNew() throws Exception {
+        final String[] expected = {};
+
+        verifyWithInlineConfigParser(
+                getPath("InputClassFanOutComplexityLambdaNew.java"), expected);
```

Comment:
This test case is not correct, but I have to add it to pass the current CI. `InputClassFanOutComplexityLambdaNew.java` should NOT have 0 complexity

Reply:
Please put comment, in code, `0 violations until #14787`

Reply:
done